### PR TITLE
Add AdamW optimizer subclass that does not apply weight decay to 1D weights

### PR DESCRIPTION
**Pull request by nickwp**
It is advised not to apply weight decay to bias parameters or BatchNorm parameters [[arxiv:1812.01187](https://arxiv.org/abs/1812.01187)].

This PR just adds a subclass of AdamW does this by creating a separate param group with weight_decay set to zero for all 1D params.

### CODE REVIEW

**Review by pdeperio — APPROVED** [review]
just a quick code read (no test), if we're short on personpower for deep reviews
